python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -430,9 +423,16 @@ def quantize(model: Model,
         :return:            A TensorflowBaseModel for INC. If there is no model found, return None.
         """
         invalidInputError(approach == 'static', "Only 'static' approach is supported now.")
-        invalidInputError(y is not None or isinstance(x, tf.data.Dataset),
-                          "y can be omitted only when x is a Dataset which returns \
-                              tuples of (inputs, targets)")
+        if not isinstance(x, tf.data.Dataset) and y is None:
+            # fake label to make quantization work
+            y = range(len(x))
+        # for TensorDataset
+        from tensorflow.python.data.ops.dataset_ops import TensorDataset, TensorSliceDataset
```

Comment:
may change to:

```python
batch = next(iter(x))
if not isinstance(batch, tuple) or len(batch) == 1:
    y = tf.data.Dataset.from_tensor_slices(list(y))    # this `list(y)` is necessary
    x = tf.data.Dataset.zip(x, y)
```

It can handle more cases, not only `TensorDataset` or `TensorSliceDataset`